internal/services/apimanagement/api_management_api_tag_description_resource.go

```diff
@@ -0,0 +1,165 @@
+package apimanagement
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/apimanagement/mgmt/2021-08-01/apimanagement"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/schemaz"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceApiManagementApiTagDescription() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceApiManagementApiTagDescriptionCreateUpdate,
+		Read:   resourceApiManagementApiTagDescriptionRead,
+		Update: resourceApiManagementApiTagDescriptionCreateUpdate,
+		Delete: resourceApiManagementApiTagDescriptionDelete,
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.ApiTagDescriptionsID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+
+			"api_name": schemaz.SchemaApiManagementApiName(),
+
+			"api_management_name": schemaz.SchemaApiManagementName(),
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"tag_id": {
```

Comment:
again, this could all be
```suggestion

			"tag_id": {
```

as we can get all that from the tag id?

`id := parse.NewApiTagDescriptionsID(subscriptionId, d.Get("resource_group_name").(string), d.Get("api_management_name").(string), d.Get("api_name").(string), d.Get("tag_id").(string))`

Reply:
`tag_id` does not have `api_name`,  at least two property needed here, `tag_id` and `api_name`
https://docs.microsoft.com/en-us/rest/api/apimanagement/current-ga/api-tag-description/create-or-update?tabs=HTTP#uri-parameters